### PR TITLE
Implement `flatiter.base` property

### PR DESCRIPTION
**Pull request by takagi**
This PR implements `flatiter.base` proeprty that is a reference to the array that is iterated over.

### CODE REVIEW

File: tests/cupy_tests/indexing_tests/test_iterate.py [hunk 1]
```diff
@@ -8,6 +8,14 @@
 from cupy import testing
 
 
+@testing.gpu
+class TestFlatiter(unittest.TestCase):
+
+    def test_base(self):
+        a = cupy.zeros((2, 3, 4))
+        assert(a.flat.base is a)
```
Comment (toslunar) [inline tests/cupy_tests/indexing_tests/test_iterate.py:16]:
```suggestion
        assert a.flat.base is a
```

Comment (takagi) [reply tests/cupy_tests/indexing_tests/test_iterate.py]: Fixed!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
The CI fails due to a sphinx warning: `docstring of cupy.ndarray::more than one target found for cross-reference 'base': cupy.ndarray.base, cupy.flatiter.base`

Using the latest sphinx `2.4.4` does not raise the warning on my local, so updating the sphinx version on chainer-test is an option.

As another, we can suppress the warning by adding `suppress_warnings = ['ref:python']` to `cupy/docs/source/conf.py`, which is introduced at https://github.com/sphinx-doc/sphinx/pull/3869 . It seems that `'ref:python'` has quite general name but takes effect only for the warning above.

```
$ cd sphinx-1.8.5/sphinx
$ grep -r "subtype='python'" .
./domains/python.py:                           type='ref', subtype='python', location=node)
```

I suppose that the latter is enough here. Does it makes sense?

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Waiting https://github.com/chainer/chainer-test/pull/575 is merged.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**